sdk/servicebus/service-bus/test/atomRuleFilters.spec.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import chai from "chai";
+import chaiAsPromised from "chai-as-promised";
+import chaiExclude from "chai-exclude";
+import * as dotenv from "dotenv";
+import {
+  CorrelationRuleFilter,
+  ServiceBusReceivedMessage,
+  ServiceBusClient,
+  ServiceBusMessage,
+  SqlRuleFilter
+} from "../src";
+import { ServiceBusAdministrationClient } from "../src/serviceBusAtomManagementClient";
+import { DEFAULT_RULE_NAME } from "../src/util/constants";
+import { recreateSubscription, recreateTopic } from "./utils/managementUtils";
+import { getConnectionString } from "./utils/testutils2";
+
+chai.use(chaiAsPromised);
+chai.use(chaiExclude);
+const should = chai.should();
+
+dotenv.config();
+
+const serviceBusAtomManagementClient: ServiceBusAdministrationClient = new ServiceBusAdministrationClient(
+  getConnectionString()
+);
+
+describe("Filter messages with the rules set by the ATOM API", () => {
+  const topicName = "new-topic";
+  const subscriptionName = "new-subscription";
+  const serviceBusClient = new ServiceBusClient(getConnectionString());
+
+  beforeEach(async () => {
+    await recreateTopic(topicName);
+    await recreateSubscription(topicName, subscriptionName);
+    await serviceBusAtomManagementClient.deleteRule(topicName, subscriptionName, DEFAULT_RULE_NAME);
+  });
+
+  after(async () => {
+    await serviceBusClient.close();
+  });
+
+  async function verifyRuleFilter(
+    messageToSend: ServiceBusMessage,
+    filter: SqlRuleFilter | CorrelationRuleFilter,
+    toCheck: (msg: ServiceBusReceivedMessage) => void
+  ) {
+    await serviceBusAtomManagementClient.createRule(
+      topicName,
+      subscriptionName,
+      "rule-name",
+      filter
+    );
+    try {
+      await serviceBusClient.createSender(topicName).sendMessages(messageToSend);
+    } catch (error) {
+      console.log(error);
```

Comment:
Why do we eat this error? Is it expected to fail for some reason that's not apparent to me?

Reply:
Ah, I'll remove the try-catch. This piece of code was from when I was testing various kinds of messages to which ones fail.

Reply:
Also, for a sanity check, would it be worth it to also send a message you know should be filtered out? Otherwise your code could pass if we failed to properly set the rule.

Reply:
Sending 2 messages now, one will be filtered into the subscription, the other one won't.